### PR TITLE
Tap discover

### PR DESCRIPTION
**Pull request by JingLin0**

### CODE REVIEW

File: tap_g2crowd/__init__.py [hunk 1]
```diff
@@ -5,6 +5,14 @@
 import singer
 from singer import utils, metadata, Catalog, CatalogEntry, Schema
 
+KEY_PROPERTIES = "id"
+STREAMS = {
+    "companies": {},
```
Comment (przywartya) [inline tap_g2crowd/__init__.py]: What happens if we don't have a replication_key?

Comment (JingLin0) [reply tap_g2crowd/__init__.py]: then we need to sync by "FULL_TABLE"

File: tap_g2crowd/__init__.py [hunk 2]
```diff
@@ -25,8 +34,37 @@ def load_schemas():
...
+                metadata=mdata,
+            )
+        )
+    return Catalog(streams)
+
+
+def do_discover():
+    catalog = discover()
+    catalog_dict = catalog.to_dict()
+    json.dump(catalog_dict, sys.stdout, indent=2, sort_keys=True)
```
Comment (paddie) [inline tap_g2crowd/__init__.py]: perhaps do this as two lines instead, one `json.dumps` and one `print` to make it clear what happens. Unless you have a very good reason, like wanting to change the output of the print to something like a file or sys.stdout at runtime should you point it to sys.stdout like this. It mostly just hurts readability.

File: tap_g2crowd/__init__.py (hunk 1 above)
Comment (przywartya) [reply tap_g2crowd/__init__.py]: But what happens when there is no repliaction_key (as in the case of `companies`) and we sync INCREMENTALLY ?

Comment (JingLin0) [reply tap_g2crowd/__init__.py]: Then the data is repeated, it is wrong

**Pull request merged**